src/orion/benchmark/task/task.py

```diff
@@ -0,0 +1,298 @@
+from abc import ABC, abstractmethod
+from dataclasses import asdict, is_dataclass
+from typing import (
+    Any,
+    ClassVar,
+    Dict,
+    Optional,
+    Generic,
+    List,
+    NamedTuple,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+import hashlib
+
+import numpy as np
+import torch
+from orion.benchmark.task.base import BaseTask
+from simple_parsing import Serializable
+from torch import Tensor
+from torch.utils.data import TensorDataset
+from simple_parsing.helpers.hparams import HyperParameters
+from orion.benchmark.task.utils import dict_union
+from logging import getLogger as get_logger
+
+
+HParams = TypeVar("HParams", bound=HyperParameters)
+logger = get_logger(__name__)
+
+
+class Domain(NamedTuple):
+    lower: np.ndarray
+    upper: np.ndarray
+
+    def __contains__(self, item: Union[np.ndarray, Tensor]) -> bool:
+        if isinstance(item, Tensor):
+            item = item.cpu().numpy()
+        assert all(self.lower <= self.upper), "lower > upper?"
+        if not item.shape[-1] == self.lower.shape[-1]:
+            return False
+        return np.all(self.lower <= item) and np.all(item <= self.upper)
+
+
+class Task(BaseTask, Generic[HParams], ABC):
+    """ Simple ABC for a 'Task'.
+    
+    TODO: Ported this from the HPO-Warm-Start repo, could perhaps merge it into BaseTask
```

Comment:
Yes I believe it would be better to merge what can be merged in BaseTask. It's suspicious that a form of duplication of BaseTask would be needed like here.

Reply:
I removed this, and split off its functionality into different subclasses of a new `TaskWrapper` class.
I'll need to remove the ones I am not actually using, and add tests for whichever ones we decide to use (in particular the wrapper for adding fixed values for a given task's search space).